simulariumio/readers/simularium_binary_reader.py

```diff
@@ -151,26 +153,42 @@ def _binary_block_spatial_data(
             if index == 0:
                 result["bundleStart"] = frame_index
             frame_n_values = int(frame_lengths[index] / BINARY_SETTINGS.BYTES_PER_VALUE)
+            if binary_spatial_data:
+                data = data_as_bytes[
+                    4 * (current_frame_offset + 3) :
+                    4 * (current_frame_offset + frame_n_values)
+                ]
+            else:
+                data = list(
+                    data_as_floats[
+                        current_frame_offset + 3 : current_frame_offset + frame_n_values
+                    ]
+                )
             result["bundleData"].append(
                 {
                     "frameNumber": frame_index,
                     "time": data_as_floats[current_frame_offset + 1],
-                    "data": list(
-                        data_as_floats[
-                            current_frame_offset
-                            + 3 : current_frame_offset
-                            + frame_n_values
-                        ]
-                    ),
+                    "nAgents": data_as_ints[current_frame_offset + 2],
+                    "data": data,
                 }
             )
             current_frame_offset += frame_n_values
         return result
 
     @staticmethod
-    def load_binary(input_file: InputFileData) -> Dict[str, Any]:
+    def load_binary(
+        input_file: InputFileData, binary_spatial_data: bool = False
```

Comment:
if anyone has a better name for this bool, let me know! I was having a hard time coming up with a concise variable name for "do the resulting spatial data in binary format or not?"

Reply:
I guess more `is_binary_spatial_data`? The InputFileData is always binary since this is the binary reader, what I want this bool to represent is whether, as we are parsing the file into individual frames, we return the spatial data as binary or not

Reply:
then `return_spatial_data_as_binary` seems wordy but better.

Reply:
or `spatial_data_as_binary` if you want to get rid of one word

Reply:
how about `parse_spatial_data_as_binary`, or the negative `parse_spatial_data`

Reply:
thanks for the brainstorm, I'm going to go with `parse_spatial_data_as_binary` because it's one character shorter than `return_spatial_data_as_binary` lol